examples/pose_graph/pose_graph_synthetic.py

```diff
@@ -197,10 +197,11 @@ def run(
         linearization_cls=LINEARIZATION_MODE[cast(str, cfg.inner_optim.solver)],
         linear_solver_cls=LINEAR_SOLVER_MODE[cast(str, cfg.inner_optim.solver)],
         vectorize=cfg.inner_optim.vectorize,
+        empty_cuda_cache=cfg.inner_optim.empty_cuda_cache,
     )
 
     # Set up Theseus layer
-    theseus_optim = th.TheseusLayer(optimizer, vectorize=cfg.inner_optim.vectorize)
+    theseus_optim = th.TheseusLayer(optimizer)
```

Comment:
Add the following two lines:

`empty_cuda_cache=cfg.solver_device=='cuda' and cfg.inner_optim.solver=='dense'`
`theseus_optim = th.TheseusLayer(optimizer, empty_cuda_cache=empty_cuda_cache)`

Reply:
This can be handled in the script via the hydra command line. We don't need to add any extra logic.